knative-operator/pkg/common/util.go

```diff
@@ -103,3 +115,34 @@ func EnsureContainerMemoryLimit(s *operatorv1alpha1.CommonSpec, containerName st
 	})
 	return
 }
+
+// common function to enqueue reconcile requests for resources
+func EnqueueRequestByOwnerAnnotations(ownerNameAnnotationKey, ownerNamespaceAnnotationKey string) handler.ToRequestsFunc {
```

Comment:
If we're refactoring anyway: Maybe this could return a `handler.EventHandler` and do the wrap into the `EnqueueRequestsFromMapFunc` directly to allow us to write:

```golang
c.Watch(&source.Kind{Type: t}, common.EnqueueRequestByOwnerAnnotations(common.ServingOwnerNamespace, common.ServingOwnerName))
```

To very closely align to osdk's syntax?

Reply:
@markusthoemmes check out the latest commit. 

I didn't like the function in function in function, but my dislike is not too strong to revert the change :) I am ok with this change.